### DISCUSSION TITLE
REFACTOR: Reduce repository size - Deprecation of direct access to data files on GitHub

### DISCUSSION THREAD

**Discussion by lucasrodes** [owid/covid-19-data#2082] [Announcements]
We are deprecating various files from our dataset in an attempt to reduce the size of the repository. Some of these files are meant for internal use, but others might have been used by some users.

In this discussion, we list all the files affected along with a deprecation date. For those files that have been used by the public, we propose alternative URLs so users can keep consuming them.

### If you use our stable URLs

If you are already using our stable URLs (`covid.ourworldindata.org/data/[filename]`), this change will not affect your workflow. See [more information](https://github.com/owid/covid-19-data/tree/master/public/data#stable-urls) on stable URLs.

### Files affected
List with all the files that are affected by the deprecation. We provide _new links_ for all of them and a _deadline_ (date by which _old url_ won't be working anymore).

| file       | new url | old url | deadline* | deprecated |
|------------|-------------------|-------------------------|--------------------------|---|
| `owid-covid-data.xlsx` | **https://covid.ourworldindata.org/data/owid-covid-data.xlsx**           | ~https://github.com/owid/covid-19-data/raw/master/public/data/owid-covid-data.xlsx~                       | November 8th, 2021                  | :white_check_mark: |
| `owid-covid-latest.xlsx` | **https://covid.ourworldindata.org/data/latest/owid-covid-latest.xlsx**           | ~https://github.com/owid/covid-19-data/raw/master/public/data/latest/owid-covid-latest.xlsx~                       | November 8th, 2021                  |  :white_check_mark: |
| `subnational_cases_deaths.zip` | **https://covid.ourworldindata.org/data/jhu/subnational_cases_deaths.zip** | ~https://github.com/owid/covid-19-data/raw/master/public/data/jhu/subnational_cases_deaths.zip~ | November 17th, 2021                  | :white_check_mark: |
| `owid-covid-data.json` | **https://covid.ourworldindata.org/data/owid-covid-data.json**    | ~https://github.com/owid/covid-19-data/raw/master/public/data/owid-covid-data.json~                   | January 7th, 2022                  | :white_check_mark: |
| `owid-covid-latest.json` | **https://covid.ourworldindata.org/data/latest/owid-covid-latest.json**    | https://github.com/owid/covid-19-data/raw/master/public/data/latest/owid-covid-data.json                  | January 10th, 2022                  | :white_large_square: |

### File candidates
List with files that might also be deprecated from GitHub.

| file       | current url |
| ---| ---| 
| `owid-covid-data.csv` | https://github.com/owid/covid-19-data/raw/master/public/data/owid-covid-data.csv     |
| `owid-covid-latest.csv` | https://github.com/owid/covid-19-data/raw/master/public/data/latest/owid-covid-latest.csv     |
| `owid-covid-data-last-ecdc.xlsx` | https://github.com/owid/covid-19-data/raw/master/public/data/ecdc/owid-covid-data-last-ecdc.xlsx     |
| `covid-testing-latest-data-source-details.xlsx` | https://github.com/owid/covid-19-data/raw/master/public/data/testing/covid-testing-latest-data-source-details.xlsx     |
| `covid-testing-all-observations.xlsx` | https://github.com/owid/covid-19-data/raw/master/public/data/testing/covid-testing-all-observations.xlsx     |

---
*Subject to revision. Check this regularly in case you need to alter your workflow, as dates may change in the coming weeks.

**Comment by sitrucp** [comment]
I am using this population file. Will its location change as well? Thanks!!

https://raw.githubusercontent.com/owid/COVID-19-data/master/scripts/input/un/population_2020.csv

**Reply by lucasrodes** [reply]
Hi @sitrucp,
The URL for this file will not be changing.
